### PR TITLE
[CALCITE-3426] Compensate validConstant type in RexLiteral

### PR DESCRIPTION
**Pull request by xy2953396112**
try to compensate validConstant type in RexLiteral.

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rex/RexLiteral.java [hunk 1]
```diff
@@ -456,9 +456,13 @@ private static RexDigestIncludeType shouldIncludeType(Comparable value, RelDataT
    * criteria as {@link #valueMatchesType}. */
   public static boolean validConstant(Object o, Litmus litmus) {
     if (o == null
```
Comment (danny0405) [inline core/src/main/java/org/apache/calcite/rex/RexLiteral.java]: It seems that this method is only used in tests ?

**Comment by yanlin-Lynn** [comment]
notice https://github.com/apache/calcite/pull/1516 related to this.

**Comment by xy2953396112** [comment]
@danny0405  thanks for you review.

File: core/src/main/java/org/apache/calcite/rex/RexLiteral.java (hunk 1 above)
Comment (jinxing64) [reply core/src/main/java/org/apache/calcite/rex/RexLiteral.java]: I think so ~

**Comment by danny0405** [comment]
@xy2953396112 Can you fix Julian's review comments to add more positive/negative tests for the tool method ? Although it is only used for tests, more tests would make it more reliable.

**Comment by danny0405** [comment]
@xy2953396112 Could you give more tests based on Julian's suggestion ?

**Comment by xy2953396112** [comment]
> @xy2953396112 Could you give more tests based on Julian's suggestion ?

ok, i will add tests later.

**Pull request closed without merging**